src/Gyro.cpp

```diff
@@ -5,100 +5,133 @@
+// If MPU-6050 is not found then returns {0,0}.
 angle_t Gyro::getCurrentAngles()
-/* Returns roll & tilt angles from MPU-6050 device in angle_t object in degrees.
-   If MPU-6050 is not found then returns {0,0}.
-*/
 {
-    const int windowSize = 16;
     // Read the accelerometer data
     struct angle_t result;
     result.pitchAngle = 0;
     result.rollAngle = 0;
     if (!isPresent)
-        return result;     // Gyro is not available
-
-    for (int i = 0; i < windowSize; i++)
     {
-        // Execute 6 byte read from MPU6050_REG_WHO_AM_I
-        Wire.beginTransmission(MPU6050_I2C_ADDR);
-        Wire.write(MPU6050_REG_ACCEL_XOUT_H); 
-        Wire.endTransmission(false);
-        Wire.requestFrom(MPU6050_I2C_ADDR, 6, 1);  // Read 6 registers total, each axis value is stored in 2 registers
-        int16_t AcX = Wire.read() << 8 | Wire.read(); // X-axis value
-        int16_t AcY = Wire.read() << 8 | Wire.read(); // Y-axis value
-        int16_t AcZ = Wire.read() << 8 | Wire.read(); // Z-axis value
-
-        // Calculating the Pitch angle (rotation around Y-axis)
-        result.pitchAngle += ((atanf(-1 * AcX / sqrtf(powf(AcY, 2) + powf(AcZ, 2))) * 180.0f / static_cast<float>(PI)) * 2.0f) / 2.0f;
-        // Calculating the Roll angle (rotation around X-axis)
-        result.rollAngle += ((atanf(-1 * AcY / sqrtf(powf(AcX, 2) + powf(AcZ, 2))) * 180.0f / static_cast<float>(PI)) * 2.0f) / 2.0f;
-
-        delay(10);  // Decorrelate measurements
+        return result; // Gyro is not available
     }
 
-    result.pitchAngle /= windowSize;
-    result.rollAngle /= windowSize;
+    // Execute 6 byte read from MPU6050_REG_WHO_AM_I
+    Wire.beginTransmission(MPU6050_I2C_ADDR);
+    Wire.write(MPU6050_REG_ACCEL_XOUT_H);
+    Wire.endTransmission(false);
+    Wire.requestFrom(MPU6050_I2C_ADDR, 6, 1); // Read 6 registers total, each axis value is stored in 2 registers
+    const uint8_t accelXMSByte = Wire.read();
+    const uint8_t accelXLSByte = Wire.read();
+    const uint8_t accelYMSByte = Wire.read();
+    const uint8_t accelYLSByte = Wire.read();
+    const uint8_t accelZMSByte = Wire.read();
+    const uint8_t accelZLSByte = Wire.read();
+
+    const int16_t accelInX = accelXMSByte << 8 | accelXLSByte; // X-axis value
+    const int16_t accelInY = accelYMSByte << 8 | accelYLSByte; // Y-axis value
+    const int16_t accelInZ = accelZMSByte << 8 | accelZLSByte; // Z-axis value
+
+    // Calculating the Pitch angle (rotation around Y-axis)
+    result.pitchAngle += ((atanf(-1 * accelInX / sqrtf(powf(accelInY, 2) + powf(accelInZ, 2))) * 180.0f / static_cast<float>(PI)) * 2.0f) / 2.0f;
+
+    // Calculating the Roll angle (rotation around X-axis)
+    result.rollAngle += ((atanf(-1 * accelInY / sqrtf(powf(accelInX, 2) + powf(accelInZ, 2))) * 180.0f / static_cast<float>(PI)) * 2.0f) / 2.0f;
+
+#if USE_GYRO_WITH_SOFTWAREI2C == 0
+    // if (Wire.getWireTimeoutFlag())
```

Comment:
Delete unused debug code

Reply:
This is not debug code. There is a library that supports timeouts

Reply:
Why is it commented out then? As far as I'm concerned, if code is commented out it's debug.

Reply:
There is a library upgrade coming where this will be used. I don't want to have to think about getting this code back once it's included.

Reply:
Ideally we would implement that code when the functionality is needed, but I can understand why you want to leave it now. Can you at least add a `// TODO` comment explaining why the code is unused?